.github/workflows/CD-pipeline-dev.yml

```diff
@@ -13,8 +13,41 @@ env:
   TEST_SETS: test-set1 test-set2 test-set3
 
 jobs:
+  check-changes:
+    runs-on: self-hosted
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Check the PR for a changes since last commit
+        id: change
+        run: |
+          SERVICES=( ${{ env.SERVICES }} )
+          git fetch origin master
+          for i in "$(git diff  --name-only HEAD^ HEAD)"
+          do
+            if [[ $i == *"internal/"* ]] || [[ $i == *"proto/"* ]] || [[ $i == "go.mod" ]] || [[ $i == "go.sum" ]] && [[ $i != *".md"* ]]; then
+              echo "RUN_TESTS=true" >> $GITHUB_OUTPUT
+              break
+            elif [[ $i == *"manifests/"* ]] && [[ $i != *".md"* ]]; then
+              echo "RUN_TESTS=true" >> $GITHUB_OUTPUT
+              break
+            else
+              for SERVICE in "${SERVICES[@]}"; do
+                if [[ $i == *"services/$SERVICE"* ]] && [[ $i != *".md"* ]]; then
+                echo "RUN_TESTS=true" >> $GITHUB_OUTPUT
+                break
+                fi
+              done
+            fi
+          done
+    outputs:
+      RUN_TESTS: ${{ steps.change.outputs.RUN_TESTS }}
+  #--------------------------------------------------------------------------------------------------
   deploy-and-monitor:
-    if: github.event.pull_request.draft == false
```

Comment:
Hi @MiroslavRepka , I see we're getting rid of this, but I don't see a "replacement" anywhere, is it intended ?

Reply:
Yes, I deleted it as it was completely unnecessary condition. The CD is triggered on `push` to `master`, thus this event does not even exist as the PR is closed at that point.